main.go

```diff
@@ -79,23 +79,20 @@ func main() {
 
 	sources := source.NewMultiSource(source.LookupMultiple(cfg.Sources...)...)
 
-	googleProvider, err := dnsprovider.NewGoogleProvider(cfg.GoogleProject, cfg.DryRun)
-	if err != nil {
-		log.Fatal(err)
-	}
+	dnsprovider.Register("google", dnsprovider.NewGoogleProvider(cfg.GoogleProject, cfg.DryRun))
+	dnsprovider.Register("aws", dnsprovider.NewAWSProvider(cfg.DryRun))
```

Comment:
yes passing "arbitrary config" to `Initialize` method is not best approach IMO. Another way, might be to allow `store` return DNS provider with/without client. But registration should occur via passing raw object. E.g:
 
```
dnsprovider.Register("aws", &dnsprovider.AWSProvider{DryRun:DryRun})
...
```
then store 
```
dnsprovider.LookUpWithClient(cfg.Dnsprovider) //creates client internally based on dns provider
```

not sure if this is any better